File: flink-app/src/main/java/com/example/ctr/infrastructure/flink/sink/DuckDBSink.java
package com.example.ctr.infrastructure.flink.sink;

import com.example.ctr.config.DuckDBProperties;
import com.example.ctr.domain.model.CTRResult;
import com.example.ctr.infrastructure.flink.sink.impl.DuckDBRichSink;
import org.apache.flink.streaming.api.functions.sink.RichSinkFunction;

public class DuckDBSink {

    private final DuckDBProperties duckDBProperties;

    public DuckDBSink(DuckDBProperties duckDBProperties) {
        this.duckDBProperties = duckDBProperties;
    }

    public RichSinkFunction<CTRResult> createSink() {
        return new DuckDBRichSink(duckDBProperties.getUrl());
    }
}

File: flink-app/src/main/kotlin/com/example/ctr/infrastructure/flink/sink/DuckDBSink.kt
package com.example.ctr.infrastructure.flink.sink

import com.example.ctr.config.DuckDBProperties
import com.example.ctr.domain.model.CTRResult
import com.example.ctr.infrastructure.flink.sink.impl.DuckDBRichSink
import org.apache.flink.streaming.api.functions.sink.RichSinkFunction

class DuckDBSink(private val duckDBProperties: DuckDBProperties) {

    fun createSink(): RichSinkFunction<CTRResult> = DuckDBRichSink(duckDBProperties.url)
}
